Applications/FileIO/Legacy/createSurface.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ *
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <cstdio>
+#include <list>
+#include <memory>
+
+#include <logog/include/logog.hpp>
+
+#include "createSurface.h"
+
+#include "Applications/FileIO/Gmsh/GMSHInterface.h"
+#include "Applications/FileIO/Gmsh/GmshReader.h"
+
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/Point.h"
+#include "GeoLib/Polygon.h"
+#include "GeoLib/Polyline.h"
+#include "GeoLib/Surface.h"
+
+#include "MeshLib/convertMeshToGeo.h"
+#include "MeshLib/Mesh.h"
+
+namespace FileIO
+{
+void createSurface(GeoLib::Polyline const& ply,
+                   GeoLib::GEOObjects& geometries,
+                   std::string const& geometry_name)
+{
+    if (!ply.isClosed())
+    {
+        WARN("Error in createSurface() - Polyline is not closed.");
+        return;
+    }
+
+    if (ply.getNumberOfPoints() <= 2)
+    {
+        WARN(
+            "Error in createSurface() - Polyline consists of less "
+            "than three points and therefore cannot be triangulated.");
+        return;
+    }
+
+    // create new GEOObjects and insert a copy of the polyline
+    auto polyline_points = std::make_unique<std::vector<GeoLib::Point*>>();
+    GeoLib::GEOObjects geo;
+    auto ply_points = ply.getPointsVec();
+    for (auto p : ply_points)
+        polyline_points->push_back(new GeoLib::Point(*p));
+    std::string ply_name = "test_ply";
```

Comment:
Changed to `temporary_polyline_name` to reflect better that this is a temporary name used only in this function.